internal/provider/resources/deployment.go

```diff
@@ -924,15 +1007,29 @@ func (r *DeploymentResource) ImportState(ctx context.Context, req resource.Impor
 // be more concise when defining the conflicting attributes. Defining them in
 // ConfigValidators instead would be much more verbose, and disconnected from
 // the source of truth.
-func validatorsForConflictingAttributes(attributes []string) []validator.String {
+func pathExpressionsForAttributes(attributes []string) []path.Expression {
```

Comment:
Refactored this slightly - it was set up to always return string validators, but now that we have IncludeSubmodules, we need bool validators as well. So I abstracted this one to return only path expressions, then two small helpers below that wrap this function and return the desired validator type.